main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -112,35 +112,29 @@ object Namer extends Phase[WeededAst.Program, NamedAst.Root] {
           val sym = Symbol.mkEnumSym(ns0, ident)
 
           // Compute the type parameters.
-          val tparams = tparams0 match {
-            case TypeParams.Elided => Nil // TODO: Support type parameter elision?
-            case TypeParams.Explicit(tps) => tps map {
-              case p =>
-                // We use a kind variable since we do not know the kind of the type variable.
-                val tvar = Type.freshVar(Kind.freshVar())
-                NamedAst.TypeParam(p, tvar, loc)
-            }
-          }
-
-          // Compute the kind of the enum.
-          val kind = Kind.mkArrow(tparams.length)
-
-          val tenv = tparams.map(kv => kv.name.name -> kv.tpe).toMap
-          val quantifiers = tparams.map(_.tpe).map(x => NamedAst.Type.Var(x, loc))
-          val enumType = if (quantifiers.isEmpty)
-            NamedAst.Type.Enum(sym, kind)
-          else {
-            val base = NamedAst.Type.Enum(sym, kind)
-            quantifiers.foldLeft(base: NamedAst.Type) {
-              case (tacc, tvar) => NamedAst.Type.Apply(tacc, tvar, loc)
-            }
-          }
+          flatMapN(getTypeParamsFromCases(tparams0, cases.values.toList, loc)) {
```

Comment:
What is happening here? Are we now allowing type parameter elision for enums? E.g.:

```
enum Foo {
  case Bar
  case Baz(a)
}
```

?